src/main/java/common/tileentities/GTMTE_LapotronicSuperCapacitor.java

```diff
@@ -314,14 +340,16 @@ public boolean checkMachine(IGregTechTileEntity thisController, ItemStack guiSlo
 
 		// Calculate total capacity
 		capacity = BigInteger.ZERO;
-		for(int i = 0; i < capacitors.length; i++){
-			if(i <= 3){
-				final long c = (long) (100000000L * Math.pow(10, i));
-				capacity = capacity.add(
-						BigInteger.valueOf(c).multiply(BigInteger.valueOf(capacitors[i])));
-			} else {
-				capacity = capacity.add(
-						MAX_LONG.multiply(BigInteger.valueOf(capacitors[i])));
+		for(int i = 0; i < capacitors.length; i++){	
+			switch(i) {
```

Comment:
You shouldn't use switches in functions. Wrap them please in a higher order function.

Reply:
Why? It works fine?

Also, I don't understand what a HOF is, even after using google and reading some articles. Somehow it's different from a normal function? And there's a first one too? Also, every example used an array, which I don't really understand too well either, so I guess it needs that too? So I don't think I'd be able to do that.